### ISSUE TITLE
Accessibility issues for blind people when creating an automation by selecting days of the week

### ISSUE THREAD

**Issue by jmortizsilva** [home-assistant/frontend#20993]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

I am a blind person and use a screen reader to operate my computer. I work with both Jaws and NVDA.

When creating an automation to run on Fridays and Saturdays, for example, it is impossible to select the days using only the keyboard. The control is not a standard drop-down and selecting the days is only possible using the mouse.

### Describe the behavior you expected

The check boxes for selecting the different days of the week should be focusable using the keyboard, it should be possible to move through them using the tab key and select them by pressing the space bar.

### Steps to reproduce the issue

1. create a new empty automation from scratch.
2. add a trigger of type time and place.
3. Choose at a specific time or on a specific date.
4. In fixed time type for example 22:00.
5. Add a condition of type time and place and choose “If the current time is before or after a specific time”.
6. In this screen is where you must select the days of the week on which you want to run this automation.

### What version of Home Assistant Core has the issue?

2024.5.5

### In which browser are you experiencing the issue with?

Microsoft Edge Versión 125.0.2535.79

### Which operating system are you using to run this browser?

Windowss 11, the last version

**Comment by karwosts** [comment]
Thank you for taking the time to report this issue. I have written a pull request that I hope will improve the situation.

**Comment by jmortizsilva** [comment]
Thanks a lot

**Comment by karwosts** [comment]
If you don't mind my asking, what path did you take to even get into the automation editor with the keyboard? Someone else just reported an issue that they could not get there with the keyboard, and I couldn't find a way either. 

I was planning to look at that too, but wanted to make sure I wasn't missing something.

**Comment by jmortizsilva** [comment]
Well, you are right, currently if only the tab key is used, it is not possible to access an automation to edit it.

 

Screen reader users use both the tab key and the arrows to move through the content. In my case, I scroll using the up and down arrows to move around the screen and press enter on the name of the automation, but it is true that if only the tab key is used, it is not possible to reach the name of the automation. You can only reach the switch to enable or disable the automation and the menu button for other options.

 

To be able to move with the arrows through the screen content, the virtual cursor must be enabled in Jaws or the focus mode disabled in NVDA.

 

IF you need me to do some other test or explain it in another way, just say so, I'd love to help as much as possible to improve HomeAssistant accessibility.

 

 

Translated with DeepL.com (free version)

 

De: karwosts ***@***.***> 
Enviado el: lunes, 10 de junio de 2024 18:19
Para: home-assistant/frontend ***@***.***>
CC: jmortizsilva ***@***.***>; Author ***@***.***>
Asunto: Re: [home-assistant/frontend] Accessibility issues for blind people when creating an automation by selecting days of the week (Issue #20993)

 

If you don't mind my asking, what path did you take to even get into the automation editor with the keyboard? Someone else just reported an issue that they could not get there with the keyboard, and I couldn't find a way either.

I was planning to look at that too, but wanted to make sure I wasn't missing something.

**Comment by jmortizsilva** [comment]
Although there have been some updates, the problem remains unresolved.

 

 

De: github-actions[bot] ***@***.***> 
Enviado el: domingo, 8 de septiembre de 2024 22:02
Para: home-assistant/frontend ***@***.***>
CC: jmortizsilva ***@***.***>; Author ***@***.***>
Asunto: Re: [home-assistant/frontend] Accessibility issues for blind people when creating an automation by selecting days of the week (Issue #20993)

 

There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.
Please make sure to update to the latest Home Assistant version and check if that solves the issue. Let us know if that works for you by adding a comment 👍
This issue has now been marked as stale and will be closed if no further activity occurs. Thank you for your contributions.

**Comment by karwosts** [comment]
keep
